src/storage-types/src/sources.rs

```diff
@@ -665,17 +665,15 @@ pub trait SourceConnection: Debug + Clone + PartialEq + AlterCompatible {
     /// The name of the resource in the external system (e.g kafka topic) if any
     fn external_reference(&self) -> Option<&str>;
 
-    /// The schema of this connection's key rows.
-    // This is mostly setting the stage for the subsequent PRs that will attempt to compute and
-    // typecheck subsequent stages of the pipeline using the types of the earlier stages of the
-    // pipeline.
-    fn key_desc(&self) -> RelationDesc;
+    /// Defines the key schema to use by default for this source connection type.
+    /// This will be used for the primary export of the source and as the default
+    /// pre-encoding key schema for the source.
+    fn default_key_desc(&self) -> RelationDesc;
```

Comment:
Is this a method that we will need after we migrate to the new system? The way I was envisioning that these methods would evolve was to instead return a map of `ExternalReference -> RelationDesc` but it's unclear if this fits with the new way of things

Reply:
> The way I was envisioning that these methods would evolve was to instead return a map of ExternalReference -> RelationDesc but it's unclear if this fits with the new way of things

That's not been necessary and since this trait is just implemented on the top level source 'connection' it's really only used to determine details related to the primary export, if any, of a source.

The external-reference stuff is also resolved at purification time, on a per-export-basis, and not necessary beyond that. I actually just realized that the `external_reference()` method on this trait is not used by anything so I will remove it since it's likely just confusing.

> Is this a method that we will need after we migrate to the new system?

Yes and no -- when we've fully migrated we will no longer need the `primary_export_details` method below since sources won't have a 'primary export'. And these `default_value_desc` and `default_key_desc` methods are essentially just returning static values that could be defined elsewhere, but seem to work well here out of convenience. They are only necessary for the source types that determine their output relations entirely based on the envelope/encoding options for each export.